### PR TITLE
server upgrade node version to 18

### PR DESCRIPTION
**Pull request by PingZhu2232**
## Description
Upgrade server packages to Node v18 as v16 is deprecated soon.

### CODE REVIEW

**Review by hedasilv — APPROVED** [review]
Hey @PingZhu2232, for my own learning, what was the test strategy for this change? r11s + Clicker or similar app, or does exporting as test packages and consuming them on the FRS side also provide test value?

**Comment by PingZhu2232** [comment]
> Hey @PingZhu2232, for my own learning, what was the test strategy for this change? r11s + Clicker or similar app, or does exporting as test packages and consuming them on the FRS side also provide test value?

I think we should consume in FRS and test with clicker to provide E2E test.

File: server/routerlicious/Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 # DisableDockerDetector "No feasible secure solution for OSS repos yet"
 
-FROM node:16.19.1-buster-slim AS base
+FROM node:18.17.1 AS base
```
Comment (znewton) [inline server/routerlicious/Dockerfile:6]: You switched these (historian and R11s) from "-slim" to not-slim. Please add the -slim back. It's important for keeping the image size down.

File: server/tinylicious/package.json [hunk 2]
```diff
@@ -86,7 +86,7 @@
 		"@types/mocha": "^9.1.1",
 		"@types/morgan": "^1.7.35",
 		"@types/nconf": "^0.10.0",
-		"@types/node": "^16.18.16",
+		"@types/node": "^18.17.1",
```
Comment (znewton) [inline server/tinylicious/package.json:89]: We're not responsible for T9s, so I'd revert these changes till the team who owns it can tackle this and do their validation.

Comment (znewton) [reply server/tinylicious/package.json:89]: Otherwise, if we do want to update T9s, then we need to do more updates, like the `engines` field in the t9s/package.json, and the t9s build pipelines. Also the T9s/devcontainer/dockerfile is still on Node 14, so we would probably want to update that as well.

Comment (znewton) [reply server/tinylicious/package.json:89]: It looks like build is failing for T9s due to these changes, so it's probably best to revert for now.

File: server/routerlicious/Dockerfile (hunk 1 above)
Comment (PingZhu2232) [reply server/routerlicious/Dockerfile]: Changed to `18.17.1-buster-slim` in all dockerfiles.

File: server/tinylicious/package.json (hunk 2 above)
Comment (PingZhu2232) [reply server/tinylicious/package.json]: Changed back.

**Review by znewton — APPROVED** [review]
